apps/web/components/dialog/ReportBookingDialog.tsx

```diff
@@ -0,0 +1,169 @@
+import type { Dispatch, SetStateAction } from "react";
+import { Controller, useForm } from "react-hook-form";
+
+import { Dialog } from "@calcom/features/components/controlled-dialog";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { ReportReason } from "@calcom/prisma/enums";
+import { trpc } from "@calcom/trpc/react";
+import { Alert } from "@calcom/ui/components/alert";
+import { Button } from "@calcom/ui/components/button";
+import { DialogContent, DialogFooter, DialogHeader } from "@calcom/ui/components/dialog";
+import { Checkbox } from "@calcom/ui/components/form";
+import { Select, Label } from "@calcom/ui/components/form";
+import { TextArea } from "@calcom/ui/components/form";
+import { showToast } from "@calcom/ui/components/toast";
+
+interface IReportBookingDialog {
+  isOpenDialog: boolean;
+  setIsOpenDialog: Dispatch<SetStateAction<boolean>>;
+  bookingId: number;
+  isRecurring: boolean;
+  isUpcoming: boolean;
+  isCancelled: boolean;
+  isRejected: boolean;
```

Comment:
Same as above, I wonder if just status makes more sense than separate props for upcoming, cancelled, or rejected here.